core/src/main/java/io/micronaut/core/util/CollectionUtils.java

```diff
@@ -189,7 +189,7 @@ public T next() {
      */
     public static <T> Set<T> setOf(T... objects) {
         if (objects == null || objects.length == 0) {
-            return Collections.emptySet();
+            return new HashSet<>();
```

Comment:
Maybe limit the size of the returned set otherwise this could change memory usage if sets are being cached somewhere